apps/web/vibes/soul/primitives/skeleton/index.tsx

```diff
@@ -0,0 +1,135 @@
+import { clsx } from 'clsx';
+
+type Props = {
+  parentClassName?: string;
+  childClassName?: string;
+  rounded?: 'sm' | 'base' | 'md' | 'lg' | 'xl' | '2xl' | '3xl' | 'full';
+} & (
+  | { type: 'text'; characterCount: number; aspectRatio?: never }
+  | { type: 'image'; aspectRatio: '5:6' | '3:4' | '1:1'; characterCount?: never }
+  | { type?: 'box' | 'circle'; characterCount?: never; aspectRatio?: never }
+);
+
+/**
+ * This component supports various CSS variables for theming. Here's a comprehensive list, along
+ * with their default values:
+ *
+ * ```css
+ * :root {
+ *   --skeleton-background: hsl(var(--contrast-100));
+ * }
+ * ```
+ */
+
+export function Skeleton({
+  parentClassName,
+  childClassName,
```

Comment:
What's the use case for exposing the `childClassName` here? It seems like the "parent" here is always just a `flex items-center` wrapper. The semantics of the "parent" vs the "child" here is unclear to me as a user of the component. I wonder if we can simplify this.

Reply:
The designs usually have a parent container that's the same height as the original component to minimize layout shift, but the actual skeleton inside typically has a smaller height than the parent container. I think the only time this is not the case is when the skeleton is for an image.

I agree that's not clear with `childClassName` and `parentClassName`. I wonder if there is a simpler approach I might not be thinking of? For example, I like that the `Skeleton.Text` is so simple because it's using `line-height` and `em` units to set the height automatically. I think we could do this with `Skeleton.Box` if we standardize how skeleton heights are set, for example, setting the height of the inner container to a percentage of the parent.